utils/eventtracking.js

```diff
@@ -90,11 +109,23 @@ const askForConsentIfNeeded = async (options) => {
     const shellSettings = settings.getShellSettings();
     // if the appropriate option is not in settings, ask now and save settings.
     if (!(TRACKING_ENABLED_KEY in shellSettings)) {
-        shellSettings[TRACKING_ENABLED_KEY] = shouldOptInByDefault() || await getEventTrackingConsent();
-        shellSettings[TRACKING_SESSION_ID_KEY] = shellSettings[TRACKING_ENABLED_KEY] ? uuid.v4() : undefined;
+        shellSettings[TRACKING_ENABLED_KEY] =
+            shouldOptInByDefault() || (await getEventTrackingConsent());
+        shellSettings[TRACKING_SESSION_ID_KEY] = shellSettings[
+            TRACKING_ENABLED_KEY
+        ]
+            ? uuid.v4()
+            : undefined;
         settings.saveShellSettings(shellSettings);
         if (shellSettings[TRACKING_ENABLED_KEY]) {
-            track(module.exports.EVENT_ID_TRACKING_OPT_IN, {}, options);
+            await track(module.exports.EVENT_ID_TRACKING_OPT_IN, {}, options);
+            await mixpanel.people.set({
```

Comment:
how does this work? If this is set globally, why is it also sent with `track`? Why call `mixpanel.people.set` after `track` and not before?

Reply:
track function is for event tracking, and people.set function is setting user profile, it does not matter for the sequence in my mind. Is that matter? @vgrichina